### PR TITLE
[flutter_workmanager] Give support for Flutter null-safety

### PR DESCRIPTION
**Pull request by parthdave93**
- Updates all versions to null-safety beta support.
- Converts dart code to give support for null-safety.
- Changed test-cases to support null-safety.

Testing instructions:

Add the following to your pubspec.yaml:

```
dependency_overrides:
  workmanager:
    git:
      url: https://github.com/parthdave93/flutter_workmanager.git
      ref: parthdave93/null_safety_migration
```

Pre-launch Checklist

  The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
  I read the Contributor Guide and followed the process outlined there for submitting PRs.
  I read the Tree Hygiene wiki page, which explains my responsibilities.
  I read and followed the Flutter Style Guide and the C++, Objective-C, Java-style guides.
  I added new tests to check the change I am making or the feature I am adding, or Hixie said the PR is tested exempt.
  I updated pubspec.yaml with an appropriate new version according to the pub versioning philosophy.
  I updated CHANGELOG.md to add a description of the change.
  I updated/added relevant documentation (doc comments with ///).
  All existing and new tests are passing.

### CODE REVIEW

**Comment by MooNag** [comment]
fixes #241

**Comment by parthdave93** [comment]
@ened can you validate the changes?

**Comment by ened** [comment]
@parthdave93 could help validate but I'm not in control of this repo, therefore can not merge it. Will need to reach out to @petermnt and @fluttercommunity/community-admins again.

**Comment by parthdave93** [comment]
Sure, I have checked the Testcases and checked with my old project in which i was using this plugin. And it works perfectly fine.
Whom should i need to connect? @ened and how to validate?

**Comment by proninyaroslav** [comment]
@parthdave93 
I tried using your fork, but it looks like it has this problem too after upgrade all of my app packages to nullsafety: https://github.com/fluttercommunity/flutter_workmanager/issues/247

```
E/AndroidRuntime( 6099): java.lang.VerifyError: Verifier rejected class androidx.work.impl.OperationImpl: com.google.common.util.concurrent.ListenableFuture androidx.work.impl.OperationImpl.getResult() failed to verify: com.google.common.util.concurrent.ListenableFuture androidx.work.impl.OperationImpl.getResult(): [0x2]  can't resolve returned type 'Unresolved Reference: com.google.common.util.concurrent.ListenableFuture' or 'Unresolved Reference: androidx.work.impl.utils.futures.SettableFuture' (declaration of 'androidx.work.impl.OperationImpl' appears in /data/app/~~UVZL9lLuZ7G6u08ofLVzBw==/com.myapp-BK3ugsfgz-p81PPUKdL3-g==/base.apk)
E/AndroidRuntime( 6099): 	at androidx.work.impl.utils.EnqueueRunnable.<init>(EnqueueRunnable.java:77)
E/AndroidRuntime( 6099): 	at androidx.work.impl.WorkContinuationImpl.enqueue(WorkContinuationImpl.java:186)
E/AndroidRuntime( 6099): 	at androidx.work.impl.WorkManagerImpl.enqueueUniqueWork(WorkManagerImpl.java:396)
E/AndroidRuntime( 6099): 	at androidx.work.WorkManager.enqueueUniqueWork(WorkManager.java:345)
E/AndroidRuntime( 6099): 	at be.tramckrijte.workmanager.WM.enqueueOneOffTask(WorkmanagerCallHandler.kt:139)
E/AndroidRuntime( 6099): 	at be.tramckrijte.workmanager.RegisterTaskHandler.enqueueOneOffTask(WorkmanagerCallHandler.kt:79)
E/AndroidRuntime( 6099): 	at be.tramckrijte.workmanager.RegisterTaskHandler.handle(WorkmanagerCallHandler.kt:57)
E/AndroidRuntime( 6099): 	at be.tramckrijte.workmanager.WorkmanagerCallHandler.handle(WorkmanagerCallHandler.kt:23)
E/AndroidRuntime( 6099): 	at be.tramckrijte.workmanager.WorkmanagerPlugin.onMethodCall(WorkmanagerPlugin.kt:42)
E/AndroidRuntime( 6099): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/AndroidRuntime( 6099): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/AndroidRuntime( 6099): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
E/AndroidRuntime( 6099): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/AndroidRuntime( 6099): 	at android.os.MessageQueue.next(MessageQueue.java:335)
E/AndroidRuntime( 6099): 	at android.os.Looper.loop(Looper.java:183)
E/AndroidRuntime( 6099): 	at android.app.ActivityThread.main(ActivityThread.java:7656)
E/AndroidRuntime( 6099): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 6099): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
E/AndroidRuntime( 6099): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by proninyaroslav** [comment]
@parthdave93 
Sorry, I found a solution. Maybe it will be useful to someone https://github.com/fluttercommunity/flutter_workmanager/issues/247#issuecomment-791502963

**Review by Ali1Ammar — APPROVED** [review]

**Review by proninyaroslav — APPROVED** [review]

File: example/pubspec.yaml
Comment (TommasoAzz) [inline example/pubspec.yaml:9]: I guess you can use 2.0.1 now.

File: example/pubspec.yaml
Comment (TommasoAzz) [inline example/pubspec.yaml:10]: And use 2.0.5 here!

File: pubspec.yaml
Comment (TommasoAzz) [inline pubspec.yaml:6]: Should it be changed to `issue_tracker: https://github.com/fluttercommunity/flutter_workmanager/issues`?
I know it works, but still

Comment (ened) [reply pubspec.yaml:6]: Yes it should!

**Comment by Nic-DB** [comment]
Hi I was having problem testing the null safety branch on IOS, I was getting the following error:
MissingPluginException(No implementation found for method initialize on channel be.tramckrijte.workmanager/foreground_channel_work_manager
I made it work by adding this lines to the pubspec.yaml under platforms:
ios:
        pluginClass: WorkmanagerPlugin

Hope it helps.

**Comment by pepe19000** [comment]
Does anyone know anything about when this PR will be merge on master repo?

**Comment by proninyaroslav** [comment]
@pepe19000 
It looks like the project has no maintainer...

**Comment by pepe19000** [comment]
> @pepe19000
> It looks like the project has no maintainer...

it does not sounds great... Is there any production solution for this (dependency_overrides is not good for me)? Maybe another well-used package?

**Comment by proninyaroslav** [comment]
> ? Maybe another well-used package?

There is a similar project, but it has a different API https://pub.dev/packages/background_fetch

**Comment by TommasoAzz** [comment]
I’m lookin too for a different package. background_fetch is maintained but it has a completely different api. On the other hand there is https://pub.dev/packages/flt_worker which has a really similar API, it is updated to BackgroundTasks on iOS (13+) but neither null-safety nor maintenance seems to exist

Scarica Outlook per iOS<https://aka.ms/o0ukef>
________________________________
Da: Yaroslav Pronin ***@***.***>
Inviato: Friday, April 9, 2021 4:33:11 PM
A: fluttercommunity/flutter_workmanager ***@***.***>
Cc: Tommaso Azzalin ***@***.***>; Comment ***@***.***>
Oggetto: Re: [fluttercommunity/flutter_workmanager] [flutter_workmanager] Give support for Flutter null-safety (#244)


? Maybe another well-used package?

There is a similar project, but it has a different API https://pub.dev/packages/background_fetch

**Comment by jeroen-meijer** [comment]
Hi all 👋🏻 

We're having somebody look at this this weekend.
Apologies for the delays.

File: example/lib/main.dart
Comment (ened) [inline example/lib/main.dart:135]: this is a unrelated change, can you please revert it?

**Review by ened — COMMENTED** [review]
Hi @parthdave93 thank you for the PR - we will probably base a future PR on this one & merge it in case you do not have time to address the feedback? Thank you again

**Comment by micimize** [comment]
You can depend on the fork like-so:
```dart
  workmanager:
    git:
      url: git@github.com:parthdave93/flutter_workmanager.git
      ref: parthdave93/null_safety_migration
```

**Comment by LostInDarkMath** [comment]
> 
> 
> Hi all 👋🏻
> 
> We're having somebody look at this this weekend.
> Apologies for the delays.

Hi, are there any news?

**Comment by jeroen-meijer** [comment]
Hey all 👋🏻

We've talked to @ened and are currently handing over control of the package and the upload process. We'll keep y'all up to date. He should have complete maintaining ownership in a few days at most.

Thanks for your patience and understanding. I wanna get this package up and running on the latest changes as fast as possible 👍🏻

**Comment by ened** [comment]
hi everyone, the project is maintained again - I have just started working my way through the open PRs and will work through the open issues as time allows. Thank you for your patience. This PR is next in the list. :)

**Pull request merged**
